File: Program/1_Array/Array Standard problem/easy/1_find_largest_three_elements.py
INT_MIN = -2**63

def find3Largest(arr):
    first, second, third = INT_MIN, INT_MIN, INT_MIN

    if len(arr) < 3:
        return -1

    for i in arr:
        if i > first:
            third = second
            second = first
            first = i
        elif i > second :
            third =second
            second = i
        else:
            third = i

    return first, second , third


l = [10,15,80,65,189,256,36,563,256,782]
print(find3Largest(l))

File: Program/1_Array/Array Standard problem/easy/1_find_largest_three_elements.c
// C program for find the largest 
// three elements in an array
#include <limits.h> /* For INT_MIN */
#include <stdio.h>

/* Function to print three largest elements */
void print3largest(int arr[], int arr_size)
{
	int i, first, second, third;

	/* There should be atleast three elements */
	if (arr_size < 3) {
		printf(" Invalid Input ");
		return;

	}

	third = second = first = INT_MIN;

    // printf("%d\n", third);
    // printf("%d\n", first);
    // printf("%d\n\n", second);

	for (i = 0; i < arr_size; i++) {
		/* If current element is greater than first*/
		if (arr[i] > first) {

			third = second;
			second = first;
			first = arr[i];
		}

		/* If arr[i] is in between first and second then update second */
		else if (arr[i] > second) {
			third = second;
			second = arr[i];
		}

		else if (arr[i] > third)
			third = arr[i];
	}

	printf("Three largest elements are %d %d %d\n", first, second, third);
}

/* Driver program to test above function */
int main()
{
	int arr[] = {3,7,10,2,5};
	int n = sizeof(arr) / sizeof(arr[0]);
	print3largest(arr, n);
	return 0;
}
//Time complexity : O(n)
